packages/workflow/src/worker-interface.ts

```diff
@@ -186,11 +186,19 @@ export function activate(activation: coresdk.workflow_activation.WorkflowActivat
         // timestamp will not be updated for activation that contain only queries
         activator.now = tsToMs(activation.timestamp);
       }
-      if (activation.historyLength == null) {
-        throw new TypeError('Got activation with no historyLength');
-      }
-      activator.info.unsafe.isReplaying = activation.isReplaying ?? false;
-      activator.info.historyLength = activation.historyLength;
+
+      // The Rust Core ensures that these activation fields are not null
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Oh... That turns out quite ugly with all of these eslint directive comments 😞

Ok, lets put back the `?? false` on booleans, and try do something about the rest. Maybe this?

```ts
// Both Core and Protobuf guarantees that these fields won't be null
activator.info.unsafe.isReplaying = activation.isReplaying ?? false;
activator.info.historyLength = activation.historyLength as number;
activator.info.continueAsNewSuggested = activation.continueAsNewSuggested ?? false;
// historySizeBytes === 0 means WFT was generated by pre-1.20.0 server
activator.info.historySizeBytes = activation.historySizeBytes?.toNumber() || undefined;
```

Reply:
Yes, too many directives...
Changing compiler directives by run-time checks is not always a good idea, but I don't think this will affect performance much...